crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/exception/GlobalExceptionHandler.java

```diff
@@ -26,7 +26,7 @@ public ProblemDetail onRuntimeException(WebClientRequestException exception){
 
     @ExceptionHandler(CrimeApplicationException.class)
     public ProblemDetail onRuntimeException(CrimeApplicationException exception){
-        ProblemDetail problemDetail = ProblemDetail.forStatusAndDetail(HttpStatus.INTERNAL_SERVER_ERROR, exception.getMessage());
+        ProblemDetail problemDetail = ProblemDetail.forStatusAndDetail(HttpStatus.NOT_FOUND, exception.getMessage());
```

Comment:
The 'CrimeApplicationException' looks to be a general purpose Exception class which now has associated with it a 404 Not Found HTTP status, we should probably do one of the following to make things easier to understand.

1. Rename `CrimeApplicationException` to `CrimeApplicationNotFoundException`
2. Add a HTTPStatus field into 'CrimeApplicationException' and use this same HTTP status in the exception handler to set the HTTP status code of the response. So we can reuse the generic `CrimeApplicationException` for different use cases (HTTP status codes).
3. Add a new Exception class specifically for this scenario and have the handler set the response code to 404 for this new exception class e.g. 'HubUsnCreationException'

Reply:
Change looks good